File: TechGarage-clean/frontend/src/services/proposalService.js
import api, { unwrap } from './api'

export const proposalService = {
  submit: (problemId, payload) => unwrap(api.post(`/problems/${problemId}/proposals`, payload)),
  getForProblem: (problemId) => unwrap(api.get(`/problems/${problemId}/proposals`)),
  getMine: () => unwrap(api.get('/proposals/mine')),
  accept: (id) => unwrap(api.put(`/proposals/${id}/accept`)),
  reject: (id) => unwrap(api.put(`/proposals/${id}/reject`)),
}

File: TechGarage-clean/backend/src/main/java/com/techgarage/service/ProposalService.java
package com.techgarage.service;

import com.techgarage.dto.proposal.ProposalRequest;
import com.techgarage.dto.proposal.ProposalResponse;
import java.util.List;

public interface ProposalService {
    ProposalResponse submit(Long problemId, ProposalRequest request);
    List<ProposalResponse> getForProblem(Long problemId);
    List<ProposalResponse> getMyProposals();
    ProposalResponse accept(Long proposalId);
    ProposalResponse reject(Long proposalId);
}
